lib/port_scanner.py

```diff
@@ -1,44 +1,50 @@
 import socket
 import threading
-from queue import Queue
 from lib.utils import Color
 
-queue = Queue()
 
-
-def scan_port(host, port):
+def scan_port(host, port, results):
     try:
         sock = socket.socket()
         sock.connect((host, port))
         RESET = Color.RESET
         GREEN = Color.GREEN
         GRAY = Color.GRAY
         BLUE = Color.BLUE
-        print(f"[{GREEN}+{RESET}] {GRAY}({host}){RESET}\tOpen port: {BLUE}{port}{RESET}")
+        print(
+            f"[{GREEN}+{RESET}] {GRAY}({host}){RESET}\tOpen port: {BLUE}{port}{RESET}"
+        )
+        current = host + "\t" + "Open port: " + str(port)
+        results.append(current)
     except:
         pass
     finally:
         sock.close()
 
 
-def scan_thread(host, t_port_start, t_port_end):
+def scan_thread(host, t_port_start, t_port_end, results):
     for port in range(t_port_start, t_port_end + 1):
-        scan_port(host, port)
+        scan_port(host, port, results)
 
 
 def scan_port_range(host, start_port, end_port, n_threads):
+    # string with results for gui version
+    results = []
```

Comment:
I'm not a fan if this results list. The idea with PixelToolkit was to also have a bunch of one file libraries which could be reused in different projects. I think we should ditch the printing in the `scan_port` function and just let the CLI part of the app get the results from this function, parse it and display to the user. The results here should be in this format: `[80, 443]` and the CLI or GUI should parse this list and respond accordingly.